serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -113,9 +110,14 @@ def add(self, request_states: list[RequestState]):
             for request_state in request_states:
                 if (
                     request_state.validation_err is not None
-                    or request_state.prompt_len >= self.max_context_length
+                    or request_state.prompt_len > min(self.max_context_length, self.max_num_batched_tokens)
```

Comment:
Now that I think about it again, should we just check with `self.max_num_batched_tokens`? In theory, the engine should be able to produce the output although its quality might be terrible.